app/src/org/commcare/provider/DebugControlsReceiver.java

```diff
@@ -0,0 +1,54 @@
+package org.commcare.provider;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+import org.commcare.CommCareApplication;
+import org.commcare.activities.LoginActivity;
+import org.commcare.android.database.global.models.ApplicationRecord;
+import org.commcare.preferences.DevSessionRestorer;
+
+/**
+ * Process broadcasts requesting to
+ * - uninstall app
+ * - save the current commcare user session.
+ * - log into the currently seated app
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class DebugControlsReceiver extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        String action = intent.getAction();
+        if (action.endsWith("SessionCaptureAction")) {
+            captureSession();
+        } else if (action.endsWith("UninstallApp")) {
+            uninstallApp(intent.getStringExtra("app_id"));
+        } else if (action.endsWith("LoginWithCreds")) {
+            login(context, intent.getStringExtra("username"), intent.getStringExtra("password"));
+        }
+    }
+
+    private static void captureSession() {
+        DevSessionRestorer.saveSessionToPrefs();
+        DevSessionRestorer.enableAutoLogin();
+    }
+
+    private static void uninstallApp(String appId) {
+        ApplicationRecord appRecord = CommCareApplication._().getAppById(appId);
+        if (appRecord != null) {
+            CommCareApplication._().expireUserSession();
+            CommCareApplication._().uninstall(appRecord);
```

Comment:
I understand why you can't call `restartCommCare()` here like the actual uninstall method does (https://github.com/dimagi/commcare-odk/blob/master/app/src/org/commcare/activities/SingleAppManagerActivity.java#L175) since you don't have an origin activity, but just a warning that not doing so could produce slightly odd behavior